buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -811,7 +811,9 @@ class BeamModulePlugin implements Plugin<Project> {
       // Provided configuration to match Maven provided scope
       project.configurations {
         provided
-        implementation.extendsFrom(provided)
+        compileOnly.extendsFrom(provided)
+        runtimeOnly.extendsFrom(provided)
+        testCompileOnly.extendsFrom(provided)
```

Comment:
We shouldn't be adding to testCompileOnly.

Dependencies that are needed during test compilation should be declared under testImplementation.
```suggestion
```